lib/extensions/attachable/active_record/base.rb

```diff
@@ -16,44 +18,51 @@ def has_many_attachments # rubocop:disable Style/PredicateName
     def has_one_attachment # rubocop:disable Style/PredicateName
       include SingularInstanceMethods
 
-      validates :attachments, length: { maximum: 1 }
+      validates :attachment_references, length: { maximum: 1 }
 
-      has_many :attachments, as: :attachable, class_name: "::#{Attachment.name}",
-                             inverse_of: :attachable, dependent: :destroy, autosave: true
+      has_many :attachment_references, as: :attachable,
```

Comment:
was there a reason why the has_many and has_many through: were not in a separate method that both has_one_attachment and has_many_attachment calls?